src/C/dense.c

```diff
@@ -2150,9 +2150,6 @@ matrix_iter(matrix *obj)
   it = PyObject_GC_New(matrixiter, &matrixiter_tp);
   if (!it) return NULL;
 
-  matrixiter_tp.tp_iter = PyObject_SelfIter;
-  matrixiter_tp.tp_getattro = PyObject_GenericGetAttr;
-
```

Comment:
This works on CPython and PyPy, but is not really supported in general. All assignments are meant to be done before calling `PyType_Ready`